azure/services/scalesets/scalesets.go

```diff
@@ -673,27 +578,57 @@ func (s *Service) getVirtualMachineScaleSetDefaultNetworkConfiguration(vmssSpec
 				})
 		}
 	}
-	return &[]compute.VirtualMachineScaleSetNetworkConfiguration{{
-		Name: pointer.String(vmssSpec.Name),
-		VirtualMachineScaleSetNetworkConfigurationProperties: &compute.VirtualMachineScaleSetNetworkConfigurationProperties{
-			Primary:            pointer.Bool(true),
-			EnableIPForwarding: pointer.Bool(true),
-			IPConfigurations: &[]compute.VirtualMachineScaleSetIPConfiguration{
-				{
-					Name: pointer.String(vmssSpec.Name),
-					VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
-						Subnet: &compute.APIEntityReference{
-							ID: pointer.String(azure.SubnetID(s.Scope.SubscriptionID(), vmssSpec.VNetResourceGroup, vmssSpec.VNetName, vmssSpec.SubnetName)),
-						},
-						Primary:                         pointer.Bool(true),
-						PrivateIPAddressVersion:         compute.IPVersionIPv4,
-						LoadBalancerBackendAddressPools: &backendAddressPools,
+	nicConfigs := []compute.VirtualMachineScaleSetNetworkConfiguration{}
+	for i, n := range vmssSpec.NetworkInterfaces {
+		nicConfig := compute.VirtualMachineScaleSetNetworkConfiguration{}
+		nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties = &compute.VirtualMachineScaleSetNetworkConfigurationProperties{}
+		nicConfig.Name = pointer.String(vmssSpec.Name + "-nic-" + strconv.Itoa(i))
+		if pointer.BoolDeref(n.AcceleratedNetworking, false) {
+			nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = pointer.Bool(true)
+		} else {
+			nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = pointer.Bool(false)
+		}
```

Comment:
on lines 438-442 we have the following code:
```
	if vmssSpec.AcceleratedNetworking == nil {
		// set accelerated networking to the capability of the VMSize
		accelNet := sku.HasCapability(resourceskus.AcceleratedNetworking)
		vmssSpec.AcceleratedNetworking = &accelNet
}
```

This sets the AcceleratedNetworking based on the instance SKU. The [comment for networkInterfaces.AcceleratedNetworking](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1beta1/types.go#L695-L696) also mentions this case. Should we use `vmssSpec.AcceleratedNetworking` if `n.AcceleratedNetworking == nil`?